data/transactions/logic/eval.go

```diff
@@ -1021,57 +996,59 @@ func opDup(cx *evalContext) {
 
 func (cx *evalContext) txnFieldToStack(txn *transactions.Transaction, field uint64) (sv stackValue, err error) {
 	err = nil
-	switch field {
-	case 0:
+	switch TxnField(field) {
+	case Sender:
 		sv.Bytes = txn.Sender[:]
-	case 1:
+	case Fee:
 		sv.Uint = txn.Fee.Raw
-	case 2:
+	case FirstValid:
 		sv.Uint = uint64(txn.FirstValid)
-	case 3:
+	case FirstValidTime:
+		sv.Uint = cx.FirstValidTimeStamp
+	case LastValid:
 		sv.Uint = uint64(txn.LastValid)
-	case 4:
+	case Note:
 		sv.Bytes = txn.Note
-	case 5:
+	case Receiver:
 		sv.Bytes = txn.Receiver[:]
-	case 6:
+	case Amount:
 		sv.Uint = txn.Amount.Raw
-	case 7:
+	case CloseRemainderTo:
 		sv.Bytes = txn.CloseRemainderTo[:]
-	case 8:
+	case VotePK:
 		sv.Bytes = txn.VotePK[:]
-	case 9:
+	case SelectionPK:
 		sv.Bytes = txn.SelectionPK[:]
-	case 10:
+	case VoteFirst:
 		sv.Uint = uint64(txn.VoteFirst)
-	case 11:
+	case VoteLast:
 		sv.Uint = uint64(txn.VoteLast)
-	case 12:
+	case VoteKeyDilution:
 		sv.Uint = txn.VoteKeyDilution
-	case 13:
+	case Type:
 		sv.Bytes = []byte(txn.Type)
-	case 14:
+	case TypeEnum:
 		sv.Uint = uint64(txnTypeIndexes[string(txn.Type)])
-	case 15:
+	case XferAsset:
 		sv.Bytes = make([]byte, 40)
 		copy(sv.Bytes, txn.XferAsset.Creator[:])
 		binary.BigEndian.PutUint64(sv.Bytes[32:], txn.XferAsset.Index)
-	case 16:
+	case AssetAmount:
 		sv.Uint = txn.AssetAmount
-	case 17:
+	case AssetSender:
 		sv.Bytes = txn.AssetSender[:]
-	case 18:
+	case AssetReceiver:
 		sv.Bytes = txn.AssetReceiver[:]
-	case 19:
+	case AssetCloseTo:
 		sv.Bytes = txn.AssetCloseTo[:]
-	case 20:
+	case GroupIndex:
 		sv.Uint = uint64(cx.GroupIndex)
-	case 21:
+	case TxID:
 		txid := txn.ID()
 		sv.Bytes = txid[:]
-	case 22:
-		sv.Uint = cx.GroupSenders[cx.GroupIndex].MicroAlgos.Raw
-	case 23:
+	/*case 22: // SenderBalance
```

Comment:
are you planning on leaving these comments in?

Reply:
cleaned up